### PR TITLE
[Materials] NorSand, fix derivatives of Lode Angle

### PR DESCRIPTION
**Pull request by ezrayst**
Fix derivatives of Lode Angle and update test cases

### CODE REVIEW

File: include/materials/norsand.tcc [hunk 1]
```diff
@@ -540,7 +568,7 @@ Eigen::Matrix<double, 6, 1> mpm::NorSand<Tdim>::compute_stress(
   (*state_vars).at("plastic_strain5") += dpstrain(5);
 
   // Update equivalent plastic deviatoric strain
-  (*state_vars).at("epds") =
+  (*state_vars).at("pdstrain") =
```
Comment (tianchiTJ) [inline include/materials/norsand.tcc:571]: I'm not sure about the coefficient in the equation of pdstrain, and I think the epds and pdstrain have different coefficients

**Review by tianchiTJ — APPROVED** [review]

File: include/materials/norsand.tcc [hunk 2]
```diff
@@ -375,19 +378,44 @@ void mpm::NorSand<Tdim>::compute_plastic_tensor(const Vector6d& stress,
...
+    // Update R
+    R = j3 / 2.0 * pow(j2 / 3.0, -1.5);
+    // Update derivatives of R
+    dR_dj2 *= pow(j2, -2.5);
+    dR_dj3 *= pow(j2, -1.5);
+    // Update derivative of theta in terms of R, check for sqrt of zero
+    if (abs(1 - pow(R, 2.0)) < 1.0E-6) {
+      dtheta_dR = 1.0 / 3.0 / sqrt(1.0E-6);
+    } else {
+      dtheta_dR = 1.0 / 3.0 / sqrt(1 - pow(R, 2.0));
```
Comment (bodhinandach) [inline include/materials/norsand.tcc]:
```suggestion
      dtheta_dR = 1.0 / 3.0 / sqrt(1 - std::pow(R, 2));
```

File: include/materials/norsand.tcc [hunk 3]
```diff
@@ -375,19 +378,44 @@ void mpm::NorSand<Tdim>::compute_plastic_tensor(const Vector6d& stress,
...
+
+  // Update when J2 is non zero
+  if (abs(j2) > 1.0E-6) {
+    // Update R
+    R = j3 / 2.0 * pow(j2 / 3.0, -1.5);
+    // Update derivatives of R
+    dR_dj2 *= pow(j2, -2.5);
+    dR_dj3 *= pow(j2, -1.5);
+    // Update derivative of theta in terms of R, check for sqrt of zero
+    if (abs(1 - pow(R, 2.0)) < 1.0E-6) {
```
Comment (bodhinandach) [inline include/materials/norsand.tcc]: Integer exponent is way faster than double or float. Please use integer if possible.
```suggestion
    if (abs(1 - pow(R, 2)) < 1.0E-6) {
```

File: include/materials/norsand.tcc (hunk 1 above)
Comment (bodhinandach) [reply include/materials/norsand.tcc:571]: Shouldn't it be the same?

**Review by bodhinandach — APPROVED** [review]
Minor comments from my side. Otherwise okay. Thanks @ezrayst

File: include/materials/norsand.tcc (hunk 1 above)
Comment (ezrayst) [reply include/materials/norsand.tcc:571]: Yes, thus we use the word `pdstrain` in this case

**Review by kks32 — APPROVED** [review]

**Pull request merged**